packages/xchain-polkadot/__mocks__/ws.ts

```diff
@@ -0,0 +1,363 @@
+// Copyright 2017-2020 @polkadot/rpc-provider authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { Server, WebSocket } from 'mock-socket';
+
+export interface Scope {
+  body: { [index: string]: Record<string, any> };
+  requests: number;
+  server: Server;
+  done: any;
+}
+
+global.WebSocket = WebSocket;
+
+// scope definition returned
+function mockWs (wsUrl: string): Scope {
+  console.log(wsUrl)
+  const server = new Server(wsUrl);
+
+  const scope: Scope = {
+    body: {},
+    done: (): void => {
+      server.stop((): void => {
+        // ignore
+      });
+    },
+    requests: 0,
+    server
+  };
+
+  server.on('connection', (socket): void => {
+    // FIXME This whole any mess is a mess
+    socket.on('message', (body: any): void => {
```

Comment:
Is this still a mess needed to be fixed :sweat_smile: ?  If not, just remove the comment ;)